gc/base/MemorySubSpace.cpp

```diff
@@ -919,16 +919,23 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 	}
 
 	if (_collector && _usesGlobalCollector) {
-		/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
-		_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
-
 		env->acquireExclusiveVMAccessForGC(_collector);
-		reportSystemGCStart(env, gcCode);
-
-		/* system GCs are accounted into "user" time in GC/total time ratio calculation */
-		_collector->garbageCollect(env, this, NULL, gcCode, NULL, NULL, NULL);
-
-		reportSystemGCEnd(env);
+		#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+		if (isSystemGCAllowed(env, gcCode)) {
+		#endif
+			/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
+			_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
+
+			reportSystemGCStart(env, gcCode);
+			/* system GCs are accounted into "user" time in GC/total time ratio calculation */
+			_collector->garbageCollect(env, this, NULL, gcCode, NULL, NULL, NULL);
+			reportSystemGCEnd(env);
+		#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+		}
```

Comment:
if bool would be generally defined there is no need to have this bracket under \#if defined(OMR_GC_IDLE_HEAP_MANAGER)